File: chapter09/repartition.py
from pyspark import SparkContext

sc = SparkContext()

arr = [i for i in range(100)]
rdd = sc.parallelize(arr)

rdd.getNumPartitions()

rdd1 = rdd.repartition(2)
rdd1.getNumPartitions()

rdd2 = rdd.repartition(4)
rdd2.getNumPartitions()

File: basic_spark/chapter09/repartition.scala
// 生成0到99的整型数组
val arr = (0 until 100).toArray
// 使用parallelize生成RDD
val rdd = sc.parallelize(arr)
 
rdd.partitions.length
// 4
 
val rdd1 = rdd.repartition(2)
rdd1.partitions.length
// 2
 
val rdd2 = rdd.repartition(8)
rdd2.partitions.length
// 8